Mathlib/AlgebraicGeometry/ProjectiveSpectrum/Basic.lean

```diff
@@ -524,4 +524,70 @@ lemma fromOfGlobalSections_toSpecZero
 
 end ofGlobalSection
 
+section nonvanishingLocus
+
+open _root_.ProjectiveSpectrum
+
+/-- Given a subset `s : Set A`, the non-vanishing locus of `s` is the set of points whose
+corresponding prime ideal does not fully contain `s`. In other words, where not all elements of `s`
+vanish. -/
+def nonvanishingLocus (s : Set A) : (Proj 𝒜).Opens :=
+  ⟨(zeroLocus 𝒜 s)ᶜ, (isClosed_zeroLocus 𝒜 s).isOpen_compl⟩
+
+lemma mem_nonvanishingLocus_iff_not_subset {s : Set A} {x : Proj 𝒜} :
+    x ∈ nonvanishingLocus 𝒜 s ↔ ¬s ⊆ x.asHomogeneousIdeal :=
+  Iff.rfl
+
+lemma mem_nonvanishingLocus_iff {s : Set A} {x : Proj 𝒜} :
+    x ∈ nonvanishingLocus 𝒜 s ↔ ∃ r ∈ s, r ∉ x.asHomogeneousIdeal := by
+  simp_rw [mem_nonvanishingLocus_iff_not_subset, Set.not_subset, SetLike.mem_coe]
+
+-- #30334
```

Comment:
Do you intend to merge this comment? If so, please expand it, so that you are not the only one that understands its meaning.